File: Projekt1/Java/ReversiMain.java
import java.util.Scanner;

public class ReversiMain {
	public static void main(String[] args) {
	    //set up initial GameState
	    GameState Game = new GameState();

	    Scanner s = new Scanner(System.in);
	    System.out.println("Enter your color (Black or White): ");
	    String player = s.next();
	    while (!player.equals("Black") && !player.equals("White")) {
	    	System.out.println("Invalid choice, please choose Black or White (mind the capital letters)");
	    	player = s.next();
	    }

	    Player p = player.equals("Black") ? Player.BLACK : Player.WHITE;
	    Game.print();
	    while (true) {
	        if (!Game.anyViableMove()) break;

	        if (Game.getCurrentPlayer() == p) {
	            // minimax(&Game, 2, moveBuffer);
	            // printf("I suggest -%s-\n", moveBuffer);
	        	String move = s.next();
	            Game.play(move);
	            // printf("Computer plays -%s-\n", moveBuffer);
	            // play(&Game, moveBuffer);
	        } else {
	            String move = ReversiAI.minimax(Game, 2);
	            System.out.println("Computer plays -" + move + "-");
	            Game.play(move);
	        }
	        Game.print();
	    }
	    String winner = (Game.blackDisks() > Game.whiteDisks()) ? "Black" : "White";
	    System.out.println(winner + " wins!");
	    s.close();
	}

}

File: Projekt1/C/ReversiMain.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "Reversi.h"

//kompileringskommando: gcc -o reversi ReversiMain.c ReversiLogic.c ReversiAI.c
int main(void) {
    //set up initial GameState
    struct GameState Game;
    resetGame(&Game);

    char input[2];
    printf("Enter your color (Black or White): \n");
    char pChar[6];
    while (scanf("%s", pChar) != 1 ||
        (strcmp(pChar, "Black") != 0 && strcmp(pChar, "White") != 0) ) {
            printf("Invalid choice, please choose Black or White (mind the capital letters): \n");
    }

    Player p = (strcmp(pChar, "Black") == 0) ? Black : White;
    char moveBuffer[3];
    printBoard(&Game);
    while (1) {
        if (!anyViableMove(&Game)) break;

        if (Game.currentP == p) {
            // minimax(&Game, 2, moveBuffer);
            // printf("I suggest -%s-\n", moveBuffer);
            if (scanf("%s", input) != 1)
                break;
            play(&Game, input);
            // printf("Computer plays -%s-\n", moveBuffer);
            // play(&Game, moveBuffer);
        } else {
            minimax(&Game, 7, moveBuffer);
            printf("Computer plays -%s-\n", moveBuffer);
            play(&Game, moveBuffer);
        }
        printBoard(&Game);
    }
    char *winner = (Game.black > Game.white) ? "Black" : "White";
    printf("%s wins!\n", winner);
    scanf("%s", input);
    return 0;
}
